Install Kubernetes v1.19.3 with minikube failed in Github actions

<!--- Please include the "minikube start" command you used in your reproduction steps --->
The error message is `invalid checksum: no checksum found in: https://storage.googleapis.com/kubernetes-release/release/v1.19.3/bin/linux/amd64/kubectl.sha256` but the hash file do exist. It works well when installing Kubernetes v1.19.2.

**Steps to reproduce the issue:** 

OS: ubuntu 16.04

Log url: https://github.com/mars-project/mars/runs/1256818740?check_suite_focus=true

Install script:

```bash
#!/bin/bash
set -e
export CHANGE_MINIKUBE_NONE_USER=true

sudo apt-get -q update || true
sudo apt-get install -yq conntrack jq

get_latest_release() {
  curl --silent "https://api.github.com/repos/$1/releases" |
    jq -c '[.[] | select(.prerelease == false)][0].tag_name' |
    sed -E 's/.*"([^"]+)".*/\1/'
}

K8S_VERSION=$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)
if [[ "$K8S_VERSION" == *"alpha"* ]] || [[ "$K8S_VERSION" == *"beta"* ]] || [[ "$K8S_VERSION" == *"rc"* ]]; then
  K8S_VERSION=$(get_latest_release "kubernetes/kubernetes")
fi

curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/$K8S_VERSION/bin/linux/amd64/kubectl && \
  chmod +x kubectl && sudo mv kubectl /usr/local/bin/

curl -Lo minikube https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64 && \
  chmod +x minikube && sudo mv minikube /usr/local/bin/

# fails here
sudo minikube start --vm-driver=none --kubernetes-version=$K8S_VERSION
sudo chown -R $(id -u):$(id -g) $HOME/.minikube
sudo chown -R $(id -u):$(id -g) $HOME/.kube

minikube update-context

JSONPATH='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}'
until kubectl get nodes -o jsonpath="$JSONPATH" 2>&1 | grep -q "Ready=True"; do
  sleep 1
done
```

<!--- TIP: Add the "--alsologtostderr" flag to the command-line for more logs --->
**Full output of failed command:** 

```
* minikube v1.14.0 on Ubuntu 16.04
* Using the none driver based on user configuration
* Starting control plane node minikube in cluster minikube
* Running on localhost (CPUs=2, Memory=6969MB, Disk=85174MB) ...
* OS release is Ubuntu 16.04.7 LTS
* Preparing Kubernetes v1.19.3 on Docker 19.03.13+azure ...

    > kubectl.sha256: 64 B / 64 B [--------------------------] 100.00% ? p/s 0s    > kubeadm.sha256: 64 B / 64 B [--------------------------] 100.00% ? p/s 0s    > kubelet.sha256: 64 B / 64 B [--------------------------] 100.00% ? p/s 0sX Exiting due to K8S_INSTALL_FAILED: updating control plane: downloading binaries: downloading kubectl: download failed: https://storage.googleapis.com/kubernetes-release/release/v1.19.3/bin/linux/amd64/kubectl?checksum=file:https://storage.googleapis.com/kubernetes-release/release/v1.19.3/bin/linux/amd64/kubectl.sha256: getter: &{Ctx:context.Background Src:https://storage.googleapis.com/kubernetes-release/release/v1.19.3/bin/linux/amd64/kubectl?checksum=file:https://storage.googleapis.com/kubernetes-release/release/v1.19.3/bin/linux/amd64/kubectl.sha256 Dst:/home/runner/.minikube/cache/linux/v1.19.3/kubectl.download Pwd: Mode:2 Detectors:[0x2d333e0 0x2d333e0 0x2d333e0 0x2d333e0 0x2d333e0 0x2d333e0] Decompressors:map[bz2:0x2d333e0 gz:0x2d333e0 tar.bz2:0x2d333e0 tar.gz:0x2d333e0 tar.xz:0x2d333e0 tbz2:0x2d333e0 tgz:0x2d333e0 txz:0x2d333e0 xz:0x2d333e0 zip:0x2d333e0] Getters:map[file:0xc000b77300 http:0xc000b87a20 https:0xc000b87a40] Dir:false ProgressListener:0x2cfd9e0 Options:[0xc8ec80]}: invalid checksum: no checksum found in: https://storage.googleapis.com/kubernetes-release/release/v1.19.3/bin/linux/amd64/kubectl.sha256
* 
* If the above advice does not help, please let us know: 
  - https://github.com/kubernetes/minikube/issues/new/choose

Error: Process completed with exit code 100.
```

**Full output of `minikube start` command used, if not already included:**



**Optional: Full output of `minikube logs` command:**
<details>


</details>

The change in hashicorp/go-getter#285 was tested with the following:

0. add a unit-test for the download function
    save [download_test.go](https://github.com/kubernetes/minikube/files/5408395/download_test.go.txt) as `./pkg/minikube/download/download_test.go`
1. vendor all pkgs
    ```
    $ go mod vendor
    ```
1. build a unit-test executable
    ```
    $ go test -mod=vendor -c -o e2e-unit.download ./pkg/minikube/download 
    ```
1. run the unit-test, it will FAIL
    ```
    $ ./e2e-unit.download -test.v
    === RUN   TestDownloadOK
    I1020 13:45:42.185907 2840778 download.go:78] Downloading: https://storage.googleapis.com/kubernetes-release/release/v1.19.2/bin/linux/amd64/kubectl?checksum=file:https://storage.googleapis.com/kubernetes-release/release/v1.19.2/bin/linux/amd64/kubectl.sha256 -> /tmp/minikube.test.download
        > kubectl.sha256: 65 B / 65 B [--------------------------] 100.00% ? p/s 0s
        > kubectl: 41.01 MiB / 41.01 MiB [---------------] 100.00% 22.26 MiB p/s 2s
    --- PASS: TestDownloadOK (2.60s)
    === RUN   TestDownloadFails
    I1020 13:45:44.785290 2840778 download.go:78] Downloading: https://storage.googleapis.com/kubernetes-release/release/v1.19.3/bin/linux/amd64/kubectl?checksum=file:https://storage.googleapis.com/kubernetes-release/release/v1.19.3/bin/linux/amd64/kubectl.sha256 -> /tmp/minikube.test.download
        > kubectl.sha256: 64 B / 64 B [--------------------------] 100.00% ? p/s 0s
        download_test.go:22: failed to download getter: &{Ctx:context.Background Src:https://storage.googleapis.com/kubernetes-release/release/v1.19.3/bin/linux/amd64/kubectl?checksum=file:https://storage.googleapis.com/kubernetes-release/release/v1.19.3/bin/linux/amd64/kubectl.sha256 Dst:/tmp/minikube.test.download.download Pwd: Mode:2 Detectors:[0x19eb018 0x19eb018 0x19eb018 0x19eb018 0x19eb018 0x19eb018] Decompressors:map[bz2:0x19eb018 gz:0x19eb018 tar.bz2:0x19eb018 tar.gz:0x19eb018 tar.xz:0x19eb018 tbz2:0x19eb018 tgz:0x19eb018 txz:0x19eb018 xz:0x19eb018 zip:0x19eb018] Getters:map[file:0xc0005327c0 http:0xc00046c3e0 https:0xc00046c400] Dir:false ProgressListener:0x19bcdd0 Options:[0xc042b0]}: invalid checksum: no checksum found in: https://storage.googleapis.com/kubernetes-release/release/v1.19.3/bin/linux/amd64/kubectl.sha256
    --- FAIL: TestDownloadFails (0.42s)
    FAIL
    ```
1. make the [change to go-getter](https://github.com/hashicorp/go-getter/pull/285/files)
    ```
    $ vim vendor/github.com/hashicorp/go-getter/checksum.go
    ```
1. build a patched unit-test executable
    ```
    $ go test -mod=vendor -c -o e2e-unit.download.patched ./pkg/minikube/download 
    ```
1. run the unit-test again, it should now pass
    ```
    $ ./e2e-unit.download.patched -test.v
    === RUN   TestDownloadOK
    I1020 13:25:42.961301 2838001 download.go:78] Downloading: https://storage.googleapis.com/kubernetes-release/release/v1.19.2/bin/linux/amd64/kubectl?checksum=file:https://storage.googleapis.com/kubernetes-release/release/v1.19.2/bin/linux/amd64/kubectl.sha256 -> /tmp/minikube.test.download
         > kubectl.sha256: 65 B / 65 B [--------------------------] 100.00% ? p/s 0s
         > kubectl: 41.01 MiB / 41.01 MiB [---------------] 100.00% 22.25 MiB p/s 2s
    --- PASS: TestDownloadOK (2.61s)
    === RUN   TestDownloadFails
    I1020 13:25:45.567012 2838001 download.go:78] Downloading: https://storage.googleapis.com/kubernetes-release/release/v1.19.3/bin/linux/amd64/kubectl?checksum=file:https://storage.googleapis.com/kubernetes-release/release/v1.19.3/bin/linux/amd64/kubectl.sha256 -> /tmp/minikube.test.download
        > kubectl.sha256: 64 B / 64 B [--------------------------] 100.00% ? p/s 0s
        > kubectl: 41.01 MiB / 41.01 MiB [---------------] 100.00% 22.33 MiB p/s 2s
    --- PASS: TestDownloadFails (2.55s)
    PASS
    ```

--- a/go.mod
+++ b/go.mod
@@ -31,7 +31,7 @@ require (
 	github.com/google/slowjam v0.0.0-20200530021616-df27e642fe7b
 	github.com/google/uuid v1.1.1
 	github.com/googleapis/gnostic v0.3.0 // indirect
-	github.com/hashicorp/go-getter v1.4.0
+	github.com/hashicorp/go-getter v1.4.2
 	github.com/hashicorp/go-retryablehttp v0.6.6
 	github.com/hooklift/assert v0.0.0-20170704181755-9d1defd6d214 // indirect
 	github.com/hooklift/iso9660 v0.0.0-20170318115843-1cf07e5970d8
@@ -96,7 +96,7 @@ replace (
 	github.com/docker/docker => github.com/docker/docker v1.4.2-0.20190924003213-a8608b5b67c7
 	github.com/docker/machine => github.com/machine-drivers/machine v0.7.1-0.20200810185219-7d42fed1b770
 	github.com/google/go-containerregistry => github.com/afbjorklund/go-containerregistry v0.0.0-20200902152226-fbad78ec2813
-	github.com/hashicorp/go-getter => github.com/afbjorklund/go-getter v1.4.1-0.20190910175809-eb9f6c26742c
+	github.com/hashicorp/go-getter => github.com/afbjorklund/go-getter v1.4.1-0.20201020145846-c0da14b4bffe
 	github.com/samalba/dockerclient => github.com/sayboras/dockerclient v1.0.0
 	k8s.io/api => k8s.io/api v0.17.3
 	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.17.3

